lib/runtime/ppx_mysql_runtime.ml

```diff
@@ -1,3 +1,57 @@
+type column_error =
+  [ `Expected_non_null_column of int * string
+  | `Deserialization_error of int * string * string * string * string ]
+
+type 'a deserializer = string -> ('a, [`Deserialization_error of string]) result
+
+let wrap_failure of_string s =
+  match of_string s with
+  | v ->
+      Ok v
+  | exception Failure _ ->
+      Error (`Deserialization_error "cannot parse number")
+
+let string_of_string str = Ok str
+
+let int_of_string = wrap_failure Pervasives.int_of_string
+
+let int32_of_string = wrap_failure Int32.of_string
+
+let int64_of_string = wrap_failure Int64.of_string
+
+let bool_of_string str =
+  match Pervasives.int_of_string str with
+  | v ->
+      Ok (v <> 0)
```

Comment:
I'd try to avoid polymorphic compare here.

Reply:
It's not obvious to me the type of comparison that the compiler will use. I would've guessed it uses a specialised integer comparison, because it does have the type information to make that decision.